File: du-backend/src/main/java/com/dwin/du/api/entity/Loan.java
package com.dwin.du.api.entity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Date;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "loan")
public class Loan {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "person_id")
    private Person person;

    @Column(name = "date")
    private Date date;

    @Column(name = "amount")
    private Double amount;

    @Column(name = "is_lent")
    private boolean lent;

    @Column(name = "is_settled")
    private boolean settled;
}

File: divvyup/DivvyUp.Shared/Models/Loan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DivvyUp_Shared.Models
{
    public class Loan
    {
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [ForeignKey("Person")]
        [Column("person_id")]
        public int PersonId { get; set; }
        public Person Person { get; set; }

        [Required]
        [Column("date")]
        public DateOnly Date { get; set; }

        [Column("amount")]
        public decimal Amount { get; set; }


        [Column("is_lent")]
        public bool Lent { get; set; }

        [Column("is_settled")]
        public bool Settled { get; set; }
    }
}
